egs/sre08/v1/sid/nnet3/xvector/allocate_egs_seg_v1.py

```diff
@@ -0,0 +1,255 @@
+
+The number of total frames is frames(feat.scp) * num_repeat.
+So frmaes(feats_li.scp) == (1/kinds_of_length) * frames(feats.scp) * num_repeat
+
+We caculate how many segment should be generated for each utterance. And then
+we average the utterance and randomly select the startpoint in each segment.
+So more than 50% frames in each utterance will be used for each new directory
+in theory. If the kinds_of_length > 2, all the frames in original directory
+should be covered.
+"""
+
+from __future__ import print_function
+import re, os, argparse, sys, math, warnings, random
+
+def get_args():
+    parser = argparse.ArgumentParser(description="In previous xvector setup, "
+            "the different examples in the same archive are equvalent length."
+            "Now, we hope the length of different examples in the same archive is "
+            "different. This script modifies the feature.scp in original data directory "
+            "to generate some new feats.scp which corresponds "
+            "to a specific length. After then, we can combine the new files, "
+            "shuffle the list and split it into pieces. At last, we use "
+            "nnet3-xvector-get-egs-seg to dump egs. The format of new feats_li.scp is "
+            "<original_uttid-startpoint-len> <extend-filename-of-features[startpoint, startpoint+len-1]>.",
+            epilog="Called by sid/nnet3/xvector/get_egs.sh")
+    parser.add_argument("--num-repeats",type=int, default=10, help="Number of "
+            "times each speaker repeats within an archive.")
+    parser.add_argument("--min-frames-per-chunk", type=int, default=50,
+            help="Minimum number of frames-per-chunk used for any archive.")
+    parser.add_argument("--max-frames-per-chunk", type=int, default=300,
+            help="Maximum number of frames-per-chunk used for any archive")
+    parser.add_argument("--kinds-of-length", type=int, default=4,
+            help="Number of length types.")
+    parser.add_argument("--randomize-chunk-length", type=str,
+            help="If true, randomly pick a chunk length in [min-frames-per-chunk, max-frames-per-chunk]."
+            "If false, the chunk length varies from min-frames-per-chunk to "
+            "max-frames-per-chunk according to a geometric sequence.",
+            default="true", choices = ["false", "true"])
+    parser.add_argument("--seed", type=int, default=123,
+            help="Seed for random number generator.")
+    parser.add_argument("--data-dir", type=str, required=True,
+            help="The location of original data directory which contains "
+            "feats.scp ,utt2num_frames and utt2spk.")
+    parser.add_argument("--output-dir", type=str, required=True,
+            help="The name of output-dir. In it, there are 'kinds-of-length' directories. "
+            "In each subdirectory, it will contains the new generated feats_li.scp and new utt2spk.")
```

Comment:
I am not sure, why you need to rename feats.scp to feats_li.scp, since it is new copy of original data directory.